database/ffldb/ldb/transaction_test.go

```diff
@@ -0,0 +1,141 @@
+package ldb
+
+import (
+	"github.com/kaspanet/kaspad/database"
+	"io/ioutil"
+	"strings"
+	"testing"
+)
+
+func TestTransactionCloseErrors(t *testing.T) {
+	tests := []struct {
+		name              string
+		function          func(dbTx *LevelDBTransaction) error
+		shouldReturnError bool
+	}{
+		{
+			name: "Put",
+			function: func(dbTx *LevelDBTransaction) error {
+				return dbTx.Put(database.MakeBucket().Key([]byte("key")), []byte("value"))
+			},
+			shouldReturnError: true,
+		},
+		{
+			name: "Get",
+			function: func(dbTx *LevelDBTransaction) error {
+				_, err := dbTx.Get(database.MakeBucket().Key([]byte("key")))
+				return err
+			},
+			shouldReturnError: true,
+		},
+		{
+			name: "Has",
+			function: func(dbTx *LevelDBTransaction) error {
+				_, err := dbTx.Has(database.MakeBucket().Key([]byte("key")))
+				return err
+			},
+			shouldReturnError: true,
+		},
+		{
+			name: "Delete",
+			function: func(dbTx *LevelDBTransaction) error {
+				return dbTx.Delete(database.MakeBucket().Key([]byte("key")))
+			},
+			shouldReturnError: true,
+		},
+		{
+			name: "Cursor",
+			function: func(dbTx *LevelDBTransaction) error {
+				_, err := dbTx.Cursor(database.MakeBucket([]byte("bucket")))
+				return err
+			},
+			shouldReturnError: true,
+		},
+		{
+			name: "Rollback",
+			function: func(dbTx *LevelDBTransaction) error {
+				return dbTx.Rollback()
+			},
+			shouldReturnError: true,
+		},
+		{
+			name: "Commit",
+			function: func(dbTx *LevelDBTransaction) error {
+				return dbTx.Commit()
+			},
+			shouldReturnError: true,
+		},
+		{
+			name: "RollbackUnlessClosed",
+			function: func(dbTx *LevelDBTransaction) error {
+				return dbTx.RollbackUnlessClosed()
+			},
+			shouldReturnError: false,
+		},
+	}
+
+	for _, test := range tests {
+		func() {
+			// Open a test db
+			path, err := ioutil.TempDir("", "TestTransactionCloseErrors")
+			if err != nil {
+				t.Fatalf("TestTransactionCloseErrors: TempDir unexpectedly "+
+					"failed: %s", err)
+			}
+			db, err := NewLevelDB(path)
+			if err != nil {
+				t.Fatalf("TestTransactionCloseErrors: Open "+
+					"unexpectedly failed: %s", err)
+			}
+			defer func() {
+				err := db.Close()
+				if err != nil {
+					t.Fatalf("TestTransactionCloseErrors: Close "+
+						"unexpectedly failed: %s", err)
+				}
+			}()
+
+			// Begin a new transaction
+			dbTx, err := db.Begin()
+			if err != nil {
+				t.Fatalf("TestTransactionCloseErrors: Begin "+
+					"unexpectedly failed: %s", err)
+			}
+			defer func() {
+				err := dbTx.RollbackUnlessClosed()
+				if err != nil {
+					t.Fatalf("TestTransactionCloseErrors: RollbackUnlessClosed "+
+						"unexpectedly failed: %s", err)
+				}
+			}()
+
+			// Close the transaction
+			err = dbTx.Commit()
```

Comment:
It might be a good idea to repeat same test-case with dbTx.Rollback, so that we make sure that it's not something only dbTx.Commit does that makes the thing work the way it should